__tests__/app.test.js

```diff
@@ -1,164 +1,262 @@
-        return request(app)
-        .get('/api/articles/notanumber/comments')
-        .expect(400)
-        .then(({ body }) => {
-            expect(body.msg).toBe("invalid input for article_id")
-        })
-    })
+  it("should respond with 200 and empty array if id is valid but has no comments", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articleComments).toEqual([]);
+      });
+  });
+});
 
-    it("should respond with 404 and id does not exist if article_id is out of bounds", () => {
-        return request(app)
-        .get('/api/articles/50/comments')
-        .expect(404)
-        .then(({ body }) => {
-            expect(body.msg).toBe("No comments found for that id")
-        })
-    })
+describe("POST /api/articles/:article_id/comments", () => {
+  it("should accept comment object and respond with 201 and posted comment", () => {
+    const testId = 2;
+    const testComment = {
+      username: "butter_bridge",
+      body: "wow what an epic comment",
+    };
+    return request(app)
+      .post(`/api/articles/${testId}/comments`)
+      .send(testComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.postedComment).toMatchObject({
+          comment_id: expect.any(Number),
+          author: "butter_bridge",
+          body: "wow what an epic comment",
+          article_id: testId,
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+        });
+      });
+  });
+  it("should respond with 404 and error when no article exists at given id", () => {
+    const testComment = {
+      username: "butter_bridge",
+      body: "wow what an epic comment",
+    };
+    return request(app)
+      .post(`/api/articles/1000/comments`)
+      .send(testComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("No article found with that id");
+      });
+  });
+  it("should respond with 404 and error if article_id is invalid", () => {
+    const testComment = {
+      username: "not_a_user",
+      body: "wow what an epic comment",
+    };
+    return request(app)
+      .post(`/api/articles/not_a_article_id/comments`)
+      .send(testComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("invalid input for article_id");
+      });
+  });
+  it("should respond with 400 and error if commentToPost has incorrect format", () => {
+    const testCases = [
+      { invalidkey: "invalid", body: "wow what an epic comment" },
+      { body: "a perfectly valid comment body" },
+      { author: "butter_bridge" },
+    ];
+   const tests = testCases.map(testCase => {
+     return request(app)
+       .post(`/api/articles/2/comments`)
+       .send(testCase)
+       .expect(400)
+       .then(({ body }) => {
+         expect(body.msg).toBe("Invalid comment to post format");
+       });
+   });
+   return Promise.all(tests)
```

Comment:
Is there a particular reason this test in a promise.all? doesn't seem needed.

Reply:
It wasn't working without one